Encode instance type in validate config test suite names

ermanno opened the pull request:
### Description of changes
Encode instance type in validate config test suite names.

### Tests
```
➜  aws-parallelcluster-cookbook git:(develop) ✗ ./kitchen.ec2.sh validate-config list
*** Apply local environment
** KITCHEN_AWS_REGION: eu-west-1
** KITCHEN_KEY_NAME: ermann-jenkins
** KITCHEN_SSH_KEY_PATH: ~/github_sources/ParallelClusterJenkins/ermann-jenkins.pem
** KITCHEN_AVAILABILITY_ZONE: a
** KITCHEN_ARCHITECTURE: x86_64
** KITCHEN_SUBNET_ID: subnet-92d373c9
** KITCHEN_VPC_ID:
** KITCHEN_SECURITY_GROUP_ID: sg-04b12b19ea1cf1ecc
** KITCHEN_IAM_PROFILE:
** KITCHEN_LOCAL_YAML: kitchen.validate-config.yml
** KITCHEN_YAML: /Users/ermann/github_sources/aws-parallelcluster-cookbook/kitchen.ec2.yml
** KITCHEN_GLOBAL_YAML: /Users/ermann/github_sources/aws-parallelcluster-cookbook/kitchen.global.yml
kitchen list
WARNING: Nokogiri was built against libxml version 2.9.4, but has dynamically loaded 2.9.13
Deprecated configuration detected:
require_chef_omnibus
chef_omnibus_url
Run 'kitchen doctor' for details.

Top level ::CompositeIO is deprecated, require 'multipart/post' and use `Multipart::Post::CompositeReadIO` instead!
Top level ::Parts is deprecated, require 'multipart/post' and use `Multipart::Post::Parts` instead!
Instance                                             Driver  Provisioner  Verifier  Transport  Last Action    Last Error
slurm-config-HeadNode-x86-64-t2micro-alinux2         Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-HeadNode-x86-64-t2micro-rhel8           Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-HeadNode-x86-64-t2micro-centos7         Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-HeadNode-x86-64-t2micro-ubuntu1804      Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-HeadNode-x86-64-t2micro-ubuntu2004      Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-HeadNode-x86-64-t2micro-ubuntu2204      Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-HeadNode-arm64-t2micro-alinux2          Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-HeadNode-arm64-t2micro-rhel8            Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-HeadNode-arm64-t2micro-centos7          Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-HeadNode-arm64-t2micro-ubuntu1804       Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-HeadNode-arm64-t2micro-ubuntu2004       Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-HeadNode-arm64-t2micro-ubuntu2204       Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
awsbatch-config-HeadNode-x86-64-t2micro-alinux2      Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
awsbatch-config-HeadNode-x86-64-t2micro-rhel8        Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
awsbatch-config-HeadNode-x86-64-t2micro-centos7      Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
awsbatch-config-HeadNode-x86-64-t2micro-ubuntu1804   Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
awsbatch-config-HeadNode-x86-64-t2micro-ubuntu2004   Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
awsbatch-config-HeadNode-x86-64-t2micro-ubuntu2204   Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
awsbatch-config-HeadNode-arm64-t2micro-alinux2       Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
awsbatch-config-HeadNode-arm64-t2micro-rhel8         Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
awsbatch-config-HeadNode-arm64-t2micro-centos7       Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
awsbatch-config-HeadNode-arm64-t2micro-ubuntu1804    Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
awsbatch-config-HeadNode-arm64-t2micro-ubuntu2004    Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
awsbatch-config-HeadNode-arm64-t2micro-ubuntu2204    Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-ComputeFleet-x86-64-t2micro-alinux2     Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-ComputeFleet-x86-64-t2micro-rhel8       Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-ComputeFleet-x86-64-t2micro-centos7     Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-ComputeFleet-x86-64-t2micro-ubuntu1804  Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-ComputeFleet-x86-64-t2micro-ubuntu2004  Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-ComputeFleet-x86-64-t2micro-ubuntu2204  Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-ComputeFleet-arm64-t2micro-alinux2      Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-ComputeFleet-arm64-t2micro-rhel8        Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-ComputeFleet-arm64-t2micro-centos7      Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-ComputeFleet-arm64-t2micro-ubuntu1804   Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-ComputeFleet-arm64-t2micro-ubuntu2004   Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
slurm-config-ComputeFleet-arm64-t2micro-ubuntu2204   Ec2     ChefInfra    Inspec    SpeedySsh  <Not Created>  <None>
```

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

ermanno committed:
Encode instance type in validate config test suite names

enrico-usai commented on kitchen.validate-config.yml:
```diff
@@ -56,7 +56,7 @@ provisioner:
 
 suites:
-  - name: slurm_config_HeadNode_x86_64
+  - name: slurm_config_HeadNode_x86_64_<%= ENV['KITCHEN_INSTANCE_TYPE'] || 't2.micro' %>
```
did you try the case where `KITCHEN_INSTANCE_TYPE` is not set?

I'm not sure we can use a dot in the suite name, anyway it's better to use `t2micro` to keep the things aligned to the changes we did in the CICD infrastructure.

ermanno replied:
This line automatically becomes:

```
slurm-config-HeadNode-x86-64-t2micro
```

Note that `KITCHEN_INSTANCE_TYPE` is a regular instance type name, such as `t3.xlarge`, etc.

ermanno committed:
Rename validate config test names to work with GitHub Actions

This patch renames the validate config test suite names to work with GitHub
Actions, in particular the latter does not support uppercase.
